sdk/src/trace/tracer.cc

```diff
@@ -33,8 +33,30 @@ nostd::unique_ptr<trace_api::Span> Tracer::StartSpan(
     const trace_api::KeyValueIterable &attributes,
     const trace_api::StartSpanOptions &options) noexcept
 {
-  return nostd::unique_ptr<trace_api::Span>{new (std::nothrow) Span{
-      this->shared_from_this(), processor_.load(), name, attributes, options}};
+  // TODO: replace nullptr with parent context in span context
+  auto sampling_result =
+      sampler_->ShouldSample(nullptr, trace_api::TraceId(), name, options.kind, attributes);
+  if (sampling_result.decision == Decision::NOT_RECORD)
+  {
+    return nostd::unique_ptr<trace_api::Span>{new (std::nothrow)
+                                                  trace_api::NoopSpan{this->shared_from_this()}};
+  }
+  else
+  {
+    auto span = nostd::unique_ptr<trace_api::Span>{new (std::nothrow) Span{
+        this->shared_from_this(), processor_.load(), name, attributes, options}};
+
+    // if the attributes is not nullptr, add attributes to the span.
+    if (sampling_result.attributes)
+    {
+      for (auto kv : *sampling_result.attributes)
```

Comment:
We can iterate over references here:
```c
for (auto &kv : *sampling_result.attributes)
```
And we can skip the `std::move` below.

Reply:
I changed it to reference. However, the program doesn't compile if I remove `std::move` in `std::move(kv.second)`.

Reply:
Please rebase. The second argument to `SetAttribute` is now passed by reference.